spec/Section 3 -- Type System.md

```diff
@@ -263,74 +263,108 @@ TypeDefinition :
   - InputObjectTypeDefinition
 
 The fundamental unit of any GraphQL Schema is the type. There are six kinds
-of named type definitions in GraphQL, and two wrapping types.
+of *named type* and two kinds of *wrapped type* definitions in GraphQL.
 
-The most basic type is a `Scalar`. A scalar represents a primitive value, like
-a string or an integer. Oftentimes, the possible responses for a scalar field
-are enumerable. GraphQL offers an `Enum` type in those cases, where the type
-specifies the space of valid responses.
+:: A *named type* is one of the six types defined with a name: *Scalar type*,
+*Enum type*, *Object type*, *Interface type*, *Union type*,
+and *Input Object type*.
 
-Scalars and Enums form the leaves in response trees; the intermediate levels are
-`Object` types, which define a set of fields, where each field is another
-type in the system, allowing the definition of arbitrary type hierarchies.
 
-GraphQL supports two abstract types: interfaces and unions.
+### Leaf and Composite Types
```

Comment:
Good time to clarify #826 .

Reply:
Thanks for making this connection! That earlier phrasing is definitely confusing. I'll weigh in on that issue and make sure this PR covers the solution. 

In writing this change I found a number of cases where we confuse "value" and "type" and interchange them even in the same paragraph. We've also made the mistake of using "scalar" when we mean "leaf".